PowerFGT/Public/cmdb/system/interface.ps1

```diff
@@ -98,3 +98,325 @@ function Get-FGTSystemInterface {
+
+        This creates a new interface using only mandatory parameters.
+
+        .EXAMPLE
+        Add-FGTSystemInterface -name PowerFGT -type vlan -alias Alias_PowerFGT -role lan -vlan_id 10 -interface port10 -admin_access https,ping,ssh -status up $true -device_identification $true -mode static -ip 192.0.2.1 -netmask 255.255.255.0 -vdom_interface root
+
+        Create an interface named PowerFGT with alias Alias_PowerFGT, role lan with vlan id 10 on interface port10. Administrative access by https and ssh, ping authorize on ip 192.0.2.1 and state connected.
+    #>
+    [CmdletBinding(DefaultParameterSetName = 'NoIP')]
+    Param(
+        [Parameter (Mandatory = $true, Position = 1)]
+        [ValidateLength(1, 15)]
+        [string]$name,
+        [Parameter (Mandatory = $true)]
+        [ValidateSet("physical", "vlan", "aggregate", "redundant", "tunnel", "vdom-link", "loopback", "switch", "hard-switch", "vap-switch", "wl-mesh", "fext-wan", "vxlan", "hdlc", "switch-vlan", "emac-vlan")]
+        [string]$type,
+        [Parameter (Mandatory = $false)]
+        [string]$alias,
+        [Parameter (Mandatory = $true)]
+        [ValidateSet('lan', 'wan', 'dmz', 'undefined')]
+        [string]$role,
+        [Parameter (Mandatory = $true)]
+        [int]$vlan_id,
+        [Parameter (Mandatory = $true)]
+        [string]$interface,
+        [Parameter (Mandatory = $false)]
+        [ValidateSet('https', 'ping', 'fgfm', 'capwap', 'ssh', 'snmp', 'ftm', 'radius-acct', 'ftm')]
+        [string[]]$admin_access,
+        [Parameter (Mandatory = $false)]
+        [ValidateSet('up', 'down')]
+        [string]$status = "up",
+        [Parameter (Mandatory = $false)]
+        [string]$device_identification = $false,
+        [Parameter (Mandatory = $true)]
+        [ValidateSet('static', 'dhcp')]
+        [string]$mode,
+        [Parameter (ParameterSetName = 'IP', Mandatory = $true)]
+        [ValidateScript( { $_ -match [IPAddress]$_ })]
+        [string]$ip,
+        [Parameter (ParameterSetName = 'IP', Mandatory = $true)]
+        [string]$netmask,
+        [Parameter (Mandatory = $true)]
+        [string]$vdom_interface,
+        [Parameter(Mandatory = $false)]
+        [String[]]$vdom,
+        [Parameter(Mandatory = $false)]
+        [psobject]$connection = $DefaultFGTConnection
+    )
+
+    Begin {
+    }
+
+    Process {
+
+        $invokeParams = @{ }
+        if ( $PsBoundParameters.ContainsKey('vdom') ) {
+            $invokeParams.add( 'vdom', $vdom )
+        }
+
+        $uri = "api/v2/cmdb/system/interface"
+        $_interface = new-Object -TypeName PSObject
+
+        $_interface | add-member -name "name" -membertype NoteProperty -Value $name
+        $_interface | add-member -name "type" -membertype NoteProperty -Value $type
+        $_interface | add-member -name "role" -membertype NoteProperty -Value $role.ToLower()
```

Comment:
I don't see the value in this. Fortigate is the one who is picky about the case, so I decided to do a tolower before sending it to the Fortigate. This way, the users of the module can use upper case in the parameters if they want. Instead of nagging to the user that they should use 'ssh' instead of 'SSH' we just handle the downcasing in the code, I think this is more elegent and userfriendly.

Reply:
i move to IgnoreCase like other cmdlet... may be look to use ToLower if it is better...